ansible_collections/arista/avd/plugins/plugin_utils/utils/template.py

```diff
@@ -33,6 +36,16 @@ def template(template_file, template_vars, templar, searchpath):
     j2template, dummy = loader._get_file_contents(template_file_path)
     j2template = to_text(j2template)
 
-    templar.available_variables = template_vars
-    result = templar.template(j2template, convert_data=False, escape_backslashes=False)
+    cache_dir = Path(template_file_path).parent.joinpath(".j2cache")
+    # Create cache_dir if not existing. We can asume the parent dir is there since we found the template file.
+    cache_dir.mkdir(0o775, False, True)
+
+    bytecode_cache = FileSystemBytecodeCache(cache_dir)
```

Comment:
Maybe we can try to define our own pattern for the cached files? https://jinja.palletsprojects.com/en/3.1.x/api/#jinja2.FileSystemBytecodeCache 

That could explain the names you were seeing (it is using the cache key) - otherwise this is the code: https://github.com/pallets/jinja/blob/52843b5cbf635b37a82ac0b6c901921a8ee076ff/src/jinja2/bccache.py#L154

but then the clear() method is implemented so we could imagine clearing everything and recompiling everything if a jinja2 template changes https://github.com/pallets/jinja/blob/52843b5cbf635b37a82ac0b6c901921a8ee076ff/src/jinja2/bccache.py#L315

Reply:
I have tried all these options. Remember the template.py is run for every template file, so a `.clear()` in here would clear for every run, defeating the purpose of cache.
The names of the files make sense to be the checksum. It will ensure that the cache is auto-updated when something changes. We could write a more clever FileSystemBytecodeCache which would track on filenames+crc and remove the invalid variant, but it would need to have the full path in the name, to handle duplicate names (ex. `main.j2`)

Reply:
Resolving this post our discussion - the conclusion is that actually the hashed filename is used for the cache filename and so given that we don't recreate many new file / change many file names, we won't need to handle this in a much clever fashion than "delete your cache folder and run again the playbook"